rkt/trust.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build linux
+
+// implements https://github.com/coreos/rocket/issues/367
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/coreos/rocket/Godeps/_workspace/src/golang.org/x/crypto/openpgp"
+	"github.com/coreos/rocket/pkg/keystore"
+)
+
+var (
+	cmdTrust = &Command{
+		Name:    cmdTrustName,
+		Summary: "Trust a key for image verification",
+		Usage:   "[--prefix PREFIX] [--do-not-ask-confirmation] PUBKEY",
+		Run:     runTrust,
+	}
+	flagPrefix string
+	flagAccept bool
+)
+
+const (
+	cmdTrustName = "trust"
+)
+
+func init() {
+	commands = append(commands, cmdTrust)
+	cmdTrust.Flags.StringVar(&flagPrefix, "prefix", "", "prefix to limit trust to")
+	cmdTrust.Flags.BoolVar(&flagAccept, "do-not-ask-confirmation", false, "suppress interactive confirmation of key addition")
```

Comment:
I would really like to be as explicit as possible that this is discouraged usage: either `--insecure` (along the lines of fetch right now), or `--yes-i-really-know-what-i-am-doing-and-want-to-trust-this-key` (I know it sounds silly, but obnoxious is better when it comes to this imho)